grid/ipfsapi/version_tree.py

```diff
@@ -0,0 +1,81 @@
+""" This module contains an implementation of an IPFS
+version-control system, which is structured as a directed in-tree with nodes
+represented by the bytes representation of the VersionTreeNode class. """
+from typing import Optional, Iterator
+
+from grid import ipfsapi
+
+# TODO: Unit tests.
+# TODO: Do we want to store the hash on the node after it's been committed?
+class VersionTreeNode:
+    """ Thin wrapper around a piece of IPFS-versioned data and the
+    IPFS multihash of its parent. """
+    # Delimiter for serializing packed object.
+    DELIMITER = b"|"
+
+    def __init__(self,
+                 contents: bytes,
+                 parent_hash: Optional[str] = None,
+                 ipfs_client: ipfsapi.Client = None):
+        """ parent_hash is a UTF-8 IPFS multihash identifying
+        this node's parent in the version tree. If parent_hash is None,
+        this node is the root of a version tree. """
+        self.contents = contents
+        # Convert empty string to None to minimize typing bugs.
+        self.parent_hash = parent_hash.decode("utf-8") or None
+        self.ipfs_client = ipfs_client
+
+    # TODO: Should there just be a singleton IPFS client in the ipfsapi.client
+    # module?  It seems like the current architecture would require
+    # it to either be passed around all over the place or potentially go
+    # through the connection process multiple times, and I don't think
+    # multiple configurations need to be maintained for a given worker node.
+    def commit(self, ipfs_client: ipfsapi.Client = None) -> str:
```

Comment:
We'll want to expose this in other methods at some point. For example with torch, we'll usually want to commit gradients of model Parameters, either directly on the Parameter's grad or with some helper function.